test/test.js

```diff
@@ -1058,14 +1058,32 @@ test('change process.cwd() to a test\'s directory', function (t) {
 
 test('Babel require hook only applies to the test file', function (t) {
 	execCli('fixture/babel-hook.js', function (err, stdout, stderr) {
-		t.ok(/exited with a non-zero exit code/.test(stderr));
-		t.ok(/Unexpected token/.test(stdout));
+		t.ok(/Unexpected token/.test(stderr));
 		t.ok(err);
 		t.is(err.code, 1);
 		t.end();
 	});
 });
 
+test('Unhandled promises will be reported to console', function (t) {
+	execCli('fixture/loud-rejection.js', function (err, stdout, stderr) {
+		t.ok(err);
+		t.ok(/You can't handle this/.test(stderr));
+		t.ok(/1 unhandled rejection[^s]/.test(stderr));
+		t.end();
+	});
+});
+
+test('uncaught exception will be reported to console', function (t) {
+	execCli('fixture/uncaught-exception.js', function (err, stdout, stderr) {
+		t.ok(err);
+		t.ok(/Can't catch me!/.test(stderr));
+		// TODO: The promise ends up rejected, so we need to track this differently
```

Comment:
Not reject the promise, and instead fulfill it with the error information.
Eventually, I'd like to see `cli.js -> error()` callback only get called in the case of a bug (something we can't handle). In a testing framework, getting an error inside the test is an expected response, and shouldn't mean we bail.